xml_converter/src/xml_converter.cpp

```diff
@@ -158,19 +172,22 @@ void process_data(
     // All of the loaded pois and categories
     vector<Parseable*> parsed_pois;
     map<string, Category> marker_categories;
-    set<string> top_level_categories;
-    vector<string> duplicate_categories;
+    map<string, vector<vector<string>>> category_to_marker_pack_names;
+    map<string, vector<string>> duplicate_categories;
```

Comment:
Should this be a map<string, set<string>> so that a filename does not appear twice if a category is present twice for some reason?